tools/distpartitioning/initialize.py

```diff
@@ -0,0 +1,450 @@
+import os
+import sys
+import math
+import numpy as np
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from timeit import default_timer as timer
+from datetime import timedelta
+from utils import augment_node_data, augment_edge_data,\
+                  read_nodes_file, read_edges_file,\
+                  read_node_features_file, read_edge_features_file,\
+                  read_metis_partitions, read_json,\
+                  get_node_types, write_metadata_json
+from globalids import assign_shuffle_global_nids_nodes, assign_shuffle_global_nids_edges,\
+                      get_shuffle_global_nids_edges
+from gloo_wrapper import gather_metadata_json
+from convert_partition import create_dgl_object
+
+def send_node_data(rank, node_data, part_ids):
+    """ Function to send node_data to non-rank-0 processes. 
+    Parameters:
+    -----------
+    rank : integer
+        rank of the process
+    node_data : numpy ndarray
+        node_data, in the augmented form
+    part_ids : python list 
+        list of unique ranks/partition-ids
+    """
+
+    for part_id in part_ids: 
+        if part_id == rank: 
+            continue
+        
+        #extract <node_type>, <global_type_nid>, <global_nid>
```

Comment:
what is `global_type_nid`?

Reply:
As discussed in the code review meeting.